src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -250,41 +245,104 @@ protected bool TryResolveUrl(string url, out string resolvedUrl)
         protected bool TryResolveUrl(string url, out IHtmlContent resolvedUrl)
         {
             resolvedUrl = null;
-            if (url == null)
+            var start = FindRelativeStart(url);
+            if (start == -1)
             {
                 return false;
             }
 
-            var trimmedUrl = url.Trim(ValidAttributeWhitespaceChars);
+            var trimmedUrl = CreateTrimmedString(url, start);
+
+            var urlHelper = UrlHelperFactory.GetUrlHelper(ViewContext);
+            var appRelativeUrl = urlHelper.Content(trimmedUrl);
+            var postTildeSlashUrlValue = trimmedUrl.Substring(2);
```

Comment:
We could avoid this substring although it makes the `EndsWith` check a lot messier.